indigo_api/models/works.py

```diff
@@ -359,6 +350,44 @@ def as_at_date(self):
         if dates:
             return max(dates)
 
+    def all_provisions(self):
+        # a list of the element ids of the earliest PiT (in whichever language) of a work
+        # does not include Schedules as they don't have commencement dates
+        first_expression = self.expressions().first()
+        if first_expression:
+            return first_expression.all_provisions()
+        return None
+
+    def main_commencement(self):
+        main = self.commencements.filter(main=True).first()
+        if main:
+            return main
+        return None
+
+    def main_commencement_date(self):
+        main = self.main_commencement()
+        if main:
+            return main.date
+        return None
```

Comment:
A python method returns None if it doesn't have an explicit return, so it's not strictly necessary to have `return None` at the end.